lib/client/boluscalc.js

```diff
@@ -0,0 +1,767 @@
+    if (record.foods.length) {
+      var html = '<table  style="float:right;margin-right:20px;font-size:12px">';
+      var carbs = 0;
+      for (var fi=0; fi<record.foods.length; fi++) {
+        var f = record.foods[fi];
+        carbs += f.carbs * f.portions;
+        html += '<tr>';
+        html += '<td>';
+        if ($('#bc_quickpick').val() < 0) { // do not allow deleting while quickpick active
+          html += '<img style="cursor:pointer" title="Delete record" src="'+icon_remove+'" href="#" class="deleteFoodRecord" index="'+fi+'">';
+        }
+        html += '</td>';
+        html += '<td>'+ f.name + '</td>';
+        html += '<td>'+ (f.portion*f.portions).toFixed(1) + ' ' + translate(f.unit) + '</td>';
+        html += '<td>('+ (f.carbs*f.portions).toFixed(1) + ' g)</td>';
+        html += '</tr>';
+      }
+      html += '</table>';
+      $('#bc_food').html(html);
+      $('.deleteFoodRecord').click(deleteFoodRecord);
+      $('#bc_carbs').val(carbs.toFixed(0));
+      $('#bc_carbs').attr('disabled',true);
+      $('#bc_gi').css('display','none');
+      $('#bc_gicalculated').css('display','');
+      $('#bc_gicalculated').text(record.gi);
+    } else {
+      $('#bc_food').html('');
+      $('#bc_carbs').attr('disabled',false);
+      $('#bc_gi').css('display','');
+      $('#bc_gicalculated').css('display','none');
+      $('#bc_gicalculated').text('');
+    }
+
+    // Show Carbs
+    if ($('#bc_usecarbs').is(':checked')) {
+      if ($('#bc_carbs').val() === '') {
+        $('#bc_carbs').css('background-color','');
+      } else if (isNaN(parseInt($('#bc_carbs').val().replace(',','.')))) {
+        $('#bc_carbs').css('background-color','red');
+      } else {
+        $('#bc_carbs').css('background-color','');
+      }
+      $('#bc_inzulincarbs').text(record.insulincarbs.toFixed(2));
+      $('#bc_inzulincarbs').attr('title','IC: ' +  ic);
+    } else {
+      $('#bc_carbs').css('background-color','');
+      $('#bc_inzulincarbs').text('');
+      $('#bc_inzulincarbs').attr('title','');
+      $('#bc_carbs').text('');
+    }
+    
+    // Show Total
+    $('#bc_rouding').text(record.roundingcorrection.toFixed(2));
+    $('#bc_insulintotal').text(record.insulin.toFixed(2));
+    
+    // Carbs needed if too much iob
+    if (record.insulin<0) {
+      $('#bc_carbsneeded').text(record.carbsneeded+' g');
+      $('#bc_insulinover').text(record.insulin.toFixed(2));
+      $('#bc_carbsneededtr').css('display','');
+      $('#bc_insulinneededtr').css('display','none');
+    } else {
+      $('#bc_carbsneeded').text('');
+      $('#bc_insulinover').text('');
+      $('#bc_carbsneededtr').css('display','none');
+      $('#bc_insulinneededtr').css('display','');
+    }
+
+    // Show basal rate
+    var basal = client.sbx.data.profile.getTempBasal(record.eventTime);
+    $('#bc_basal').text((basal.treatment ? 'T ' : '') + basal.tempbasal.toFixed(3));
+  };
+  
+  boluscalc.gatherBoluscalcData = function gatherBoluscalcData() {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (20)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1240172972/issues/source?bid=2452379&fileBranchId=2935247#l312)